### PR TITLE
CCM-5656: update report for parallel send

### PR DESCRIPTION
**Pull request by alibombaye**
## Description

Updated the completed comms report to handle when we have a requestItem but with no plans.

Details of testing carried out for other scenarios are documented here https://nhsd-jira.digital.nhs.uk/browse/CCM-5656

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: infrastructure/terraform/components/reporting/scripts/sql/reports/completed_comms_report.sql [hunk 1]
```diff
@@ -1,5 +1,6 @@
 SELECT
     ri.requestitemid AS requestitemid,
+    ri.clientid AS clientid,
```
Comment (alexnuttall) [inline infrastructure/terraform/components/reporting/scripts/sql/reports/completed_comms_report.sql:3]: Did we decide not to add any extra fields (e.g. requestItemRefId in the end?)

Comment (mark-r-bjss) [reply infrastructure/terraform/components/reporting/scripts/sql/reports/completed_comms_report.sql:3]: We don't need the client ID in the select as it is in the path.

Comment (mark-r-bjss) [reply infrastructure/terraform/components/reporting/scripts/sql/reports/completed_comms_report.sql:3]: The ticket defines the fields we want in the select.

You are missing:
request-item.requestId
request-item.requestRefId
request-item.requestItemRefId
request-item.completedDate
request-item.failedReason
request-item.sendingGroupId
request-item.sendingGroupIdVersion
request-item-plan.failedReason

and need to remove
clientid
requestitemcreateddate,

File: infrastructure/terraform/components/reporting/scripts/sql/reports/completed_comms_report.sql [hunk 2]
```diff
@@ -13,9 +14,7 @@ LEFT OUTER JOIN request_item_plan_status rip ON
     ri.clientid = rip.clientid
 WHERE
     ri.clientid = ? AND
-    ri.status IN ('FAILED', 'DELIVERED') AND
-    rip.status IN ('FAILED', 'DELIVERED', 'SKIPPED') AND
     (
-        DATE(ri.completedtime) = DATE(?) OR DATE(rip.completedtime) = DATE(?)
+        (rip.status IN ('FAILED', 'DELIVERED', 'SKIPPED') AND DATE(rip.completedtime) = DATE(?)) OR
+        (ri.status IN ('FAILED') AND rip.status is null AND DATE(ri.completedtime) = DATE(?))
```
Comment (mark-r-bjss) [inline infrastructure/terraform/components/reporting/scripts/sql/reports/completed_comms_report.sql:19]: Minor point but I'd maybe replace `rip.status is null` with `rip. requestitemplanid is null` as an ID field should never be null (but the status may not always have the same limitation).

File: infrastructure/terraform/components/reporting/scripts/sql/reports/completed_comms_report.sql (hunk 1 above)
Comment (alibombaye) [reply infrastructure/terraform/components/reporting/scripts/sql/reports/completed_comms_report.sql:3]: Sorry I've not updated the fields just yet. Just kept in minimum for the purpose of testing. Will update the select with the fields as suggested.

**Review by mark-r-bjss — DISMISSED** [review]

File: infrastructure/terraform/components/reporting/scripts/sql/reports/completed_comms_report.sql (hunk 1 above)
Comment (alibombaye) [reply infrastructure/terraform/components/reporting/scripts/sql/reports/completed_comms_report.sql]: Updated to include additional fields

**Review by chris-elliott-nhsd — APPROVED** [review]

**Review by sidnhs — APPROVED** [review]

**Pull request merged**